src/tcldcc.c

```diff
@@ -1295,7 +1308,8 @@ static int tcl_listen STDVAR
   }
   strlcpy(type, argv[i], sizeof(type));
 /* Check if mask or proc exists */
-  if (((argc>3) && !strlen(ip)) || ((argc >4) && strlen(ip))) {
+  if ((((argc>3) && !strlen(ip)) || ((argc >4) && strlen(ip))) &&
```

Comment:
While you're touching this code, add some more improvements: ```!ip[0]``` instead of ```!strlen(ip)``` (and inverse) and use ```char ip[121] = ""``` at declaration to also get rid of the ```else``` clause at line 1287.
Lots of ```strlen```'s down below too.